text/0117-typo-tolerance-settings-api.md

```diff
@@ -0,0 +1,202 @@
+
+### 3.1.4. `minWordSizeFor1Typo`
+
+- Type: Integer
+- Required: False
+- Default: `5`
+
+`minWordSizeFor1Typo` customize the minimum size for a word to tolerate 1 typo.
+
+> Given the default value `5`, the search engine starts to apply typo tolerance on a query term if its length is at least equal to 5 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
+
+### 3.1.5. `minWordSizeFor2Typos`
+
+- Type: Integer
+- Required: False
+- Default: `9`
+
+`minWordSizeFor2Typos` customize the minimum size for a word to tolerate 2 typos.
+
+> Given the default value `9`, the search engine handles up to 2 typos on a query term if its length is at least equal to 9 characters.
+
+> See [2.1. Typos calculation section](#21-typos-calculation)
+
+## 3.2. API Endpoints Definition
+
+### 3.2.1. `indexes/:index_uid/settings/typo-tolerance`
+
+Manage the typo tolerance configuration for an index.
+
+#### 3.2.1.1. `GET` - `indexes/:index_uid/settings/typo-tolerance`
+
+Allow fetching the current definition of the typo tolerance feature for an index.
+
+`200` - Response body
+
+```json
+    {
+        "enabled": true,
+        "disableOnAttributes": [],
+        "disableOnWords": [],
+        "minWordSizeFor1Typo": 5,
+        "minWordSizeFor2Typos": 9
+    }
+```
+
+All properties must be returned when the resource is retrieved.
+
+##### 3.2.1.1.2. Errors
+
+- 🔴 If the index does not exist, the API returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error.
+- 🔴 If Meilisearch is secured, accessing this route without the `Authorization` header returns a [missing_authorization_header](0061-error-format-and-definitions.md#missing_authorization_header) error.
+- 🔴 If Meilisearch is secured, accessing this route with a key that does not have permissions (i.e. other than the master-key) returns an [invalid_api_key](0061-error-format-and-definitions.md#invalid_api_key) error.
+
+#### 3.2.1.2. `POST` - `indexes/:index_uid/settings/typo-tolerance`
+
+Allow customizing partially the default settings of the typo tolerance feature for an index.
+
+Request payload
+
+```json
+{
+    "enabled": true,
+    "disableOnAttributes": [
+        "title",
+        "description"
+    ],
+    "disableOnWords": [
+        "Tolkien"
+    ],
+    "minWordSizeFor1Typo": 4,
+    "minWordSizeFor2Typos": 8
+}
+```
+
+`202 Accepted` - Example Response
+
+```json
+{
+    "uid": 42,
+    "indexUid": "books",
+    "status": "enqueued",
+    "type": "settingsUpdate",
+    "enqueuedAt": "2022-03-01T18:39:29.228155Z"
+}
+```
+
+> Returns a 202 response. See [Summarized `task` Object for `202 Accepted`](0060-tasks-api.md#summarized-task-object-for-202-accepted)
+
+##### 3.2.1.2.1. Errors
+
+- 🔴 If the index does not exist, the API returns an [index_not_found](0061-error-format-and-definitions.md#index_not_found) error.
```

Comment:
This one is tricky. It will depend on if the instance is secure and if the API key allows the operation or not. I'll look into it to explain it better but it will add a lot of information. 

I've added that kind of details in https://github.com/meilisearch/specifications/pull/123/files. I will backport those details here when #123 is merged!

Reply:
I've backported the details in the specification.